idaes/core/solvers/tests/test_petsc.py

```diff
@@ -0,0 +1,541 @@
+    m.u = pyo.Var(m.time)
+    m.dxdt = pyodae.DerivativeVar(m.x, wrt=m.time)
+
+    def diff_eq_rule(m, t):
+        return m.dxdt[t] == m.x[t]**2 - m.u[t]
+    m.diff_eq = pyo.Constraint(m.time, rule=diff_eq_rule)
+
+    discretizer = pyo.TransformationFactory('dae.finite_difference')
+    discretizer.apply_to(m,nfe=1,scheme='BACKWARD')
+
+    for t in m.time:
+        m.x[t].fix(2.0*t)
+
+    m.u[0].fix(1.0)
+
+    return m
+
+def rp_example2():
+    m = pyo.ConcreteModel()
+
+    m.time = pyodae.ContinuousSet(initialize=(0.0, 10.0))
+    m.x = pyo.Var(m.time)
+    m.u = pyo.Var(m.time)
+    m.t = pyo.Var(m.time)
+    m.dxdt = pyodae.DerivativeVar(m.x, wrt=m.time)
+
+    def diff_eq_rule(m, t):
+        return m.dxdt[t] == m.x[t]**2 - m.u[t]
+    m.diff_eq = pyo.Constraint(m.time, rule=diff_eq_rule)
+
+    def x_eq_rule(m, t):
+        return m.x[t] == 2.0*m.t[t]
+    m.x_eq = pyo.Constraint(m.time, rule=x_eq_rule)
+
+    discretizer = pyo.TransformationFactory('dae.finite_difference')
+    discretizer.apply_to(m,nfe=1,scheme='BACKWARD')
+
+    m.u[0].fix(1.0)
+    m.t[0].fix(m.time.first())
+
+    return m
+
+def rp_example3():
+    m = pyo.ConcreteModel()
+
+    m.time = pyodae.ContinuousSet(initialize=(0.0, 10.0))
+    m.x = pyo.Var(m.time)
+    m.u = pyo.Var(m.time)
+    m.dxdt = pyodae.DerivativeVar(m.x, wrt=m.time)
+
+    def diff_eq_rule(m, t):
+        return m.dxdt[t] == m.x[t]**2 - m.u[t]
+    m.diff_eq = pyo.Constraint(m.time, rule=diff_eq_rule)
+
+    discretizer = pyo.TransformationFactory('dae.finite_difference')
+    discretizer.apply_to(m,nfe=1,scheme='BACKWARD')
+
+    m.u[0].fix(1.0)
+    m.dxdt[:].fix(2.0)
+
+    return m
+
+def rp_example4():
+    m = pyo.ConcreteModel()
+
+    m.time = pyodae.ContinuousSet(initialize=(0.0, 10.0))
+    m.x = pyo.Var(m.time, initialize=1)
+    m.u = pyo.Var(m.time, initialize=1)
+    m.dxdt = pyodae.DerivativeVar(m.x, wrt=m.time)
+
+    def diff_eq1_rule(m, t):
+        return m.dxdt[t] == m.x[t]**2 - m.u[t]
+    m.diff_eq1 = pyo.Constraint(m.time, rule=diff_eq1_rule)
+
+    def diff_eq2_rule(m, t):
+        return m.dxdt[t] == 2.0
+    m.diff_eq2 = pyo.Constraint(m.time, rule=diff_eq2_rule)
+
+    discretizer = pyo.TransformationFactory('dae.finite_difference')
+    discretizer.apply_to(m,nfe=1,scheme='BACKWARD')
+
+    m.u[0].fix(1.0)
+    m.x[0].fix(0.0)
+    m.diff_eq2[0].deactivate()
+
+    return m
+
+def rp_example5():
+    m = pyo.ConcreteModel()
+
+    m.time = pyodae.ContinuousSet(initialize=(0.0, 10.0))
+    m.x = pyo.Var(m.time, initialize=1)
+    m.u = pyo.Var(m.time, initialize=1)
+    m.dxdt = pyodae.DerivativeVar(m.x, wrt=m.time)
+
+    def diff_eq1_rule(m, t):
+        return m.dxdt[t] == m.x[t]**2 - m.u[t]
+    m.diff_eq1 = pyo.Constraint(m.time, rule=diff_eq1_rule)
+
+    discretizer = pyo.TransformationFactory('dae.finite_difference')
+    discretizer.apply_to(m,nfe=1,scheme='BACKWARD')
+
+    m.u[0].fix(1.0)
+    m.dxdt[:].fix(2)
+
+    return m
```

Comment:
What's the difference between `rp_example5` and `rp_example3`?

Reply:
You know, I don't see a difference and can't remember.  I think I must have copied it and meant to change it to test another case, and now I cannot recall what it was for.  Doing to many things at once I guess.  If the tests pass I'll merge this and make another small PR to update or remove the test if I still can't remember why it's there.